### PR TITLE
BB-1358 Add i18n tools config, transifex and automatization

### PR DESCRIPTION
**Pull request by xirdneh**
This PR adds the necessary configuration for edx's i18n tools and transifex. it also adds the necessary maketargets to maintain translations. 

**JIRA tickets**: 

**Discussions**:

**Dependencies**: None

**Screenshots**: 
Screen Shot 2019-06-14 at 11 27 17 AM

**Sandbox URL**: TBD - sandbox is being provisioned.

**Merge deadline**: "None" if there's no rush, "ASAP" if it's critical, or provide a specific date if there is one.

**Testing instructions**:

1. Checkout this branch
2. Change edx language to spanish
3. Check the messages are not in english.

**Author notes and concerns**:

- Not really sure if the transifex configuration is correct since I'm not sure how to add a repo there.

### CODE REVIEW

File: chat/conf/locale [hunk 1]
```diff
@@ -0,0 +1 @@
+../translations
```
Comment (viadanna) [inline chat/conf/locale:1]: This should be in `chat/locale`, actually a sibling of `translations`

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	@echo "Please use \`make <target>' where <target> is one of"
+	@perl -nle'print $& if m{^[a-zA-Z_-]+:.*?## .*$$}' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m  %-25s\033[0m %s\n", $$1, $$2}'
+
+upgrade:
+	pip-compile --output-file test_requirements.txt test_requirements.in
+
+extract_translations: ## extract strings to be translated, outputting .po files
+	cd $(WORKING_DIR) && i18n_tool extract
+	mv $(EXTRACTED_DJANGO) $(EXTRACTED_TEXT)
+	tail -n +20 $(EXTRACTED_DJANGOJS) >> $(EXTRACTED_TEXT)
```
Comment (viadanna) [inline Makefile]: This is currently crashing with:
```
tail: cannot open 'chat/translations/en/LC_MESSAGES/djangojs-partial.po' for reading: No such file or directory
```
Seems no tagged strings are present in static, so this should be removed.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	rm $(EXTRACTED_DJANGOJS)
+	sed -i'' -e 's/nplurals=INTEGER/nplurals=2/' $(EXTRACTED_TEXT)
+	sed -i'' -e 's/plural=EXPRESSION/plural=\(n != 1\)/' $(EXTRACTED_TEXT)
+
+compile_translations: ## compile translation files, outputting .mo files for each supported language
+	cd $(WORKING_DIR) && i18n_tool generate
+	python manage.py compilejsi18n --namespace ChatXBlockI18N --output $(JS_TARGET)
```
Comment (viadanna) [inline Makefile:27]: This step is failing as well.

File: chat/settings.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    locale_config = yaml.load(locale_config_file)
+
+    LANGUAGES = [
+        (code, code,)
+        for code in locale_config['locales'] + locale_config['dummy_locales']
+    ]
+
+STATICI18N_DOMAIN = 'textjs'
+
+# `ProblemI18N` might collide with other modules, using `ProblemBuilderXBlockI18N` to be more explicit
```
Comment (viadanna) [inline chat/settings.py]: You can remove the references to the other xblock.

**Review by viadanna — CHANGES_REQUESTED** [review]

File: chat/conf/locale (hunk 1 above)
Comment (xirdneh) [reply chat/conf/locale:1]: I thought edx i18n tools use this `chat/conf/locale` directory.
There's another symlink at the root directory with the name `locale` which also goes to translations. I think that's the one I have to move to `/chat/.`

File: Makefile (hunk 2 above)
Comment (xirdneh) [reply Makefile]: Ah, true. I missed the fact it was erroring out since it was creating the `django.po` file. Thanks.

File: Makefile (hunk 3 above)
Comment (xirdneh) [reply Makefile:27]: @viadanna This is not failing for me. Can you paste the output you're seeing?

Comment (viadanna) [reply Makefile:27]:
```
ImportError: No module named mako.template
Makefile:26: recipe for target 'compile_translations' failed
```

Sorry, should have been more specific. Just missing `mako` from the requirements.

Comment (xirdneh) [reply Makefile:27]: @viadanna Ah yeap, I did that manually but forgot to put it in the requirements. Let me add it.

**Comment by xirdneh** [comment]
@viadanna I've also removed `.mo` from `.gitignore` so we can track the compiled translations and added a screenshot showing the spanish translation.

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+help: ## display this help message
+	@echo "Please use \`make <target>' where <target> is one of"
+	@perl -nle'print $& if m{^[a-zA-Z_-]+:.*?## .*$$}' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m  %-25s\033[0m %s\n", $$1, $$2}'
+
+upgrade:
+	pip-compile --output-file test_requirements.txt test_requirements.in
+
+extract_translations: ## extract strings to be translated, outputting .po files
+	cd $(WORKING_DIR) && i18n_tool extract
+	if [ -f "$(EXTRACTED_DJANGOJS)" ]; then msgcat $(EXTRACTED_DJANGO) $(EXTRACTED_DJANGOJS) -o $(EXTRACTED_TEXT); else mv $(EXTRACTED_DJANGO) $(EXTRACTED_TEXT); fi
```
Comment (viadanna) [inline Makefile:19]: Looking good :+1:

**Review by viadanna — APPROVED** [review]
Good to go 👍

- [x] I tested this by running the `make` targets and checking the translations are correctly extracted and compiled.
- [x] I read through the code.
- [ ] ~I checked for accessibility issues.~
- [ ] ~Includes documentation.~

**Comment by viadanna** [comment]
@xirdneh Oh, wait, I almost forgot. Can you add a section to the README explaining how to use the `make` targets?

**Comment by xirdneh** [comment]
@viadanna Done, thanks!

**Pull request merged**